src/main/scala/org/ergoplatform/nodeView/state/DigestState.scala

```diff
@@ -48,27 +48,29 @@ class DigestState protected(override val version: VersionTag,
         case Some(proofs) if !java.util.Arrays.equals(ADProofs.proofDigest(proofs.proofBytes), fb.header.ADProofsRoot) =>
           Failure(new Error("Incorrect proofs digest"))
         case Some(proofs) =>
-          Try {
-            val txs = fb.blockTransactions.txs
-            val currentStateContext = stateContext.appendHeader(fb.header)
-
-            val declaredHash = fb.header.stateRoot
-            // Check modifications, returning sequence of old values
-            val oldValues: Seq[ErgoBox] = proofs.verify(ErgoState.stateChanges(txs), rootHash, declaredHash)
-              .get.map(v => ErgoBoxSerializer.parseBytes(v).get)
-            val knownBoxes = (txs.flatMap(_.outputs) ++ oldValues).map(o => (ByteArrayWrapper(o.id), o)).toMap
-            val totalCost = txs.map { tx =>
-              tx.statelessValidity.get
-              val boxesToSpend = tx.inputs.map(_.boxId).map { id =>
-                knownBoxes.get(ByteArrayWrapper(id)) match {
-                  case Some(box) => box
-                  case None => throw new Error(s"Box with id ${Algos.encode(id)} not found")
+          stateContext.appendFullBlock(fb, votingSettings).flatMap {currentStateContext =>
+            Try {
```

Comment:
You can replace `flatMap` to `map` and remove this `Try`

Reply:
done